data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -719,30 +719,31 @@ private ObjectAndMetaData<BlockRewardData> getBlockRewardData(
     }
 
     final UInt64 participantReward = spec.getSyncCommitteeParticipantReward(state);
-    final UInt64 proposerReward =
+    final long proposerReward =
         participantReward
             .times(PROPOSER_WEIGHT)
-            .dividedBy(WEIGHT_DENOMINATOR.minus(PROPOSER_WEIGHT));
-    return blockAndMetaData.map(value -> calculateBlockRewards(proposerReward, block, state));
+            .dividedBy(WEIGHT_DENOMINATOR.minus(PROPOSER_WEIGHT))
+            .longValue();
+    return blockAndMetaData.map(__ -> calculateBlockRewards(proposerReward, block, state));
   }
 
   private BlockRewardData calculateBlockRewards(
-      final UInt64 proposerReward,
+      final long proposerReward,
       final BeaconBlock block,
       final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState state) {
     final SyncAggregate aggregate = block.getBody().getOptionalSyncAggregate().orElseThrow();
 
     final int proposerIndex = spec.getBeaconProposerIndex(state, block.getSlot());
     final long attestationsBlockRewards = calculateAttestationRewards(state);
-    final UInt64 syncAggregateBlockRewards =
+    final long syncAggregateBlockRewards =
         calculateProposerSyncAggregateBlockRewards(proposerReward, aggregate);
-    final UInt64 proposerSlashingsBlockRewards = calculateProposerSlashingsRewards(block, state);
-    final UInt64 attesterSlashingsBlockRewards = calculateAttesterSlashingsRewards(block, state);
+    final long proposerSlashingsBlockRewards = calculateProposerSlashingsRewards(block, state);
+    final long attesterSlashingsBlockRewards = calculateAttesterSlashingsRewards(block, state);
     final long total =
         attestationsBlockRewards
-            + syncAggregateBlockRewards.longValue()
-            + proposerSlashingsBlockRewards.longValue()
-            + attesterSlashingsBlockRewards.longValue();
+            + syncAggregateBlockRewards
+            + proposerSlashingsBlockRewards
+            + attesterSlashingsBlockRewards;
```

Comment:
nit, could just let the BlockRewardData do the increment given its summing the fields you pass to it, and its a reasonable expectation for it to be able to do that...